extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/config/MongoEntityConfigProvider.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.mongodb.panache.config;
+
+import static org.eclipse.microprofile.config.ConfigProvider.getConfig;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.quarkus.mongodb.panache.MongoEntity;
+
+public class MongoEntityConfigProvider {
+    public static final String MONGODB_DEFAULT_DATABASE = "quarkus.mongodb.database";
+
+    private static final String MONGODB_DEFAULT_CLIENT_NAME = "";
+    private static final Map<String, String> databaseNameByClient = new ConcurrentHashMap<>();
+
+    public static String getMongoDatabaseName(MongoEntity entity) {
+        final String databaseName;
+        if (entity != null && !entity.database().isEmpty()) {
+            databaseName = entity.database();
+        } else {
+            String clientName = (entity == null) ? MONGODB_DEFAULT_CLIENT_NAME : entity.clientName();
+            databaseName = getDatabaseName(clientName);
+        }
+
+        return databaseName;
+    }
+
+    private static String getDatabaseName(String clientName) {
+        Objects.requireNonNull(clientName);
+        return databaseNameByClient.computeIfAbsent(
+                clientName,
+                c -> getConfig()
```

Comment:
`c` is a poor variable name

Reply:
I agree but with `computeIfAbsent` I'm used to name the parameter of the function like this because I known that this parameter is the same as the first parameter of `computeIfAbsent`. Anyway I added a more specific name.